admin/services/data-access/pupil-restart.data.service.js

```diff
@@ -46,10 +53,99 @@ pupilRestartDataService.findLatest = async function (options) {
 
 /**
  * Get all the restart codes documents
+ * @deprecated Please use sqlFindRestartCodes instead
  * @return {Promise.<{Object}>}
  */
 pupilRestartDataService.getRestartCodes = async () => {
   return RestartCode.find().lean().exec()
 }
 
+/** SQL METHODS **/
+
+/**
+ * Create a new pupil restart.
+ * @param {object} data
+ * @return  { insertId: <number>, rowsModified: <number> }
+ */
+pupilRestartDataService.sqlCreate = async (data) => {
+  return sqlService.create(table, data)
+}
+
+/**
+ * Returns number of restarts specified by pupil id
+ * @param pupilId
+ * @return {Promise.<*>}
+ */
+pupilRestartDataService.sqlGetNumberOfRestartsByPupil = async function (pupilId) {
+  const sql = `SELECT COUNT(*) FROM ${sqlService.adminSchema}.[pupilRestart] WHERE pupil_id=@pupilId`
+  const params = [
+    {
+      name: 'pupilId',
+      value: pupilId,
+      type: TYPES.Int
+    }
+  ]
+  return sqlService.query(sql, params)
+}
+
+/**
+ * Find latest check for pupil
+ * @param pupilId - the pupil taking the check
+ * @return {Promise.<void>} - lean Check objects
+ */
+pupilRestartDataService.sqlFindLatestRestart = async function (pupilId) {
+  const sql = `SELECT TOP 1 * FROM ${sqlService.adminSchema}.[pupilRestart] WHERE pupil_id=@pupilId AND isDeleted=0 ORDER BY createdAt DESC`
+  const params = [
+    {
+      name: 'pupilId',
+      value: pupilId,
+      type: TYPES.Int
+    }
+  ]
+  return sqlService.query(sql, params)
+}
+
+/**
+ * Find latest check for pupil
+ * @param pupilId - the pupil taking the check
+ * @return {Promise.<void>} - lean Check objects
+ */
+pupilRestartDataService.sqlFindRestartCodes = async function () {
+  const sql = `
+  SELECT 
+    id, 
+    code, 
+    statusDesc 
+  FROM ${sqlService.adminSchema}.[pupilRestartCode]
+  ORDER BY statusDesc ASC`
+  return sqlService.query(sql)
+}
+
+/**
+ * Mark an existing pupil restart as deleted
+ * @param pupilId
+ * @param userId
+ * @return {Promise<*>}
+ */
+pupilRestartDataService.sqlMarkRestartAsDeleted = async (pupilId, userId) => {
+  const params = [
+    {
+      name: 'pupilId',
+      value: pupilId,
+      type: TYPES.Int
+    },
+    {
+      name: 'deletedByUser_id',
+      value: userId,
+      type: TYPES.Int
+    },
+    {
+      name: 'updatedAt',
+      value: moment.utc(),
+      type: TYPES.DateTimeOffset
+    }
+  ]
+  return sqlService.modify(`UPDATE ${sqlService.adminSchema}.[pupilRestart] SET isDeleted=1, deletedByUser_id=@userId, updatedAt=@updatedAt WHERE [pupil_id]=@pupilId`, params)
```

Comment:
the caller appears to be expecting an object to be returned

Reply:
updated

Reply:
Please format this sql.